src/checkout-buttons/strategies/googlepay/googlepay-checkoutcom-button-strategy.spec.ts

```diff
@@ -0,0 +1,191 @@
+            new ConfigActionCreator(new ConfigRequestSender(requestSender))
+        );
+
+        paymentProcessor = createGooglePayPaymentProcessor(
+            store,
+            new GooglePayCheckoutcomInitializer()
+        );
+
+        formPoster = createFormPoster();
+
+        strategy = new GooglePayButtonStrategy(
+            store,
+            formPoster,
+            checkoutActionCreator,
+            paymentProcessor
+        );
+
+        jest.spyOn(store, 'dispatch')
+            .mockReturnValue(Promise.resolve(store.getState()));
+
+        jest.spyOn(paymentProcessor, 'initialize')
+            .mockReturnValue(Promise.resolve());
+
+        jest.spyOn(store.getState().paymentMethods, 'getPaymentMethod')
+            .mockReturnValue(paymentMethod);
+
+        jest.spyOn(formPoster, 'postForm')
+            .mockReturnValue(Promise.resolve());
+
+        container = document.createElement('div');
+        container.setAttribute('id', 'googlePayCheckoutButton');
+        walletButton = document.createElement('a');
+        walletButton.setAttribute('id', 'mockButton');
+
+        jest.spyOn(paymentProcessor, 'createButton')
+            .mockReturnValue(walletButton);
+
+        jest.spyOn(paymentProcessor, 'deinitialize');
+
+        document.body.appendChild(container);
+    });
+
+    afterEach(() => {
+        document.body.removeChild(container);
+    });
+
+    describe('#initialize()', () => {
+        describe('Payment method exist', () => {
+            it('Creates the button', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.createButton).toHaveBeenCalled();
+            });
+
+            it('Validates if strategy has been initialized', async () => {
+                checkoutButtonOptions = getCheckoutButtonOptions();
+
+                await strategy.initialize(checkoutButtonOptions);
+
+                setTimeout(() => {
+                    strategy.initialize(checkoutButtonOptions);
+                }, 0);
+
+                strategy.initialize(checkoutButtonOptions);
+
+                expect(paymentProcessor.initialize).toHaveBeenCalledTimes(1);
+            });
+
+            it('fails to initialize the strategy if no container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.UndefinedContainer);
+
+                function initialize() {
+                    strategy.initialize(checkoutButtonOptions);
+                }
+                expect(initialize).toThrow(InvalidArgumentError);
+            });
+
+            it('fails to initialize the strategy if no valid container id is supplied', () => {
+                checkoutButtonOptions = getCheckoutButtonOptions(Mode.InvalidContainer);
+
+                return expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
```

Comment:
maybe you can use async/await instead of return?
```js
await expect(strategy.initialize(checkoutButtonOptions)).rejects.toThrow(InvalidArgumentError);
```

Reply:
removed the 'return' keyword, as jest will wait fo the promise to resolve/reject on its own